### DISCUSSION TITLE
How to create a model for view which inherit from a real table?

### DISCUSSION THREAD

**Discussion by PaleNeutron** [sqlalchemy/sqlalchemy#11668] [Usage Questions] [answered]
I have already create a view with complex where condition in my database and I want to use it as normal ORM model in my code.

The base table model or the view have several columns and relationships and I want to inherit the base table model and just change `__tablename__` to the view, thus I can use it as a real table.

But the relationship back_populates are defined between real table and other tables. I'm not sure how to do this inherit.

**Comment by zzzeek** [comment, marked as answer]
you'd need to define a common superclass for the "table" and the "view" versions of things, and you'd need the `relationship()` call to be invoked separately for each of these models.    likely an approach that uses a common [declarative mixin](https://docs.sqlalchemy.org/en/20/orm/declarative_mixins.html#mixing-in-relationships), where relationships are underneath `@declared_attr`.   the `mapped_column` construct already knows how to copy itself out for different subclasses.
